src/pages/CustomerInvoicePage.js

```diff
@@ -74,120 +71,47 @@ export const CustomerInvoicePage = ({
   runWithLoadingIndicator,
   routeName,
 }) => {
-  const [tableState, dispatch, instantDebouncedDispatch] = usePageReducer(routeName, {
-    backingData: transaction.items,
-    data: transaction.items.sorted('item.name').slice(),
-    database,
-    keyExtractor,
-    dataState: new Map(),
-    currentFocusedRowKey: null,
-    searchTerm: '',
-    filterDataKeys: ['item.name'],
-    sortBy: 'itemName',
-    isAscending: true,
-  });
-
-  const [modalKey, setModalKey] = useState(null);
-  const [modalIsOpen, setModalIsOpen] = useState(false);
-  const { ITEM_SELECT, COMMENT_EDIT, THEIR_REF_EDIT } = MODAL_KEYS;
-  const { data, dataState, sortBy, isAscending, columns } = tableState;
-  let isSelection = false;
-
-  // eslint-disable-next-line no-restricted-syntax
-  for (const row of dataState.values()) {
-    if (row.isSelected) {
-      isSelection = true;
-      break;
-    }
-  }
-
-  const openItemSelector = () => {
-    setModalKey(ITEM_SELECT);
-    setModalIsOpen(true);
-  };
-
-  const openCommentEditor = () => {
-    setModalKey(COMMENT_EDIT);
-    setModalIsOpen(true);
-  };
-
-  const openTheirRefEditor = () => {
-    setModalKey(THEIR_REF_EDIT);
-    setModalIsOpen(true);
-  };
-
-  const closeModal = () => setModalIsOpen(false);
-
-  const getModalTitle = () => {
-    switch (modalKey) {
-      default:
-      case ITEM_SELECT:
-        return modalStrings.search_for_an_item_to_add;
-      case COMMENT_EDIT:
-        return modalStrings.edit_the_invoice_comment;
-      case THEIR_REF_EDIT:
-        return modalStrings.edit_their_reference;
+  const [tableState, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(
+    routeName,
+    {
+      pageObject: transaction,
+      backingData: transaction.items,
```

Comment:
This also needs to be sorted, as the when deleting items this is used again. Can either be lazy and add sorted here, or you can pass a method for initial state so you can do the sort once then just slice the result for `data`.

Reply:
hmm, after using `sorted` so much recently for vaccines stuff and knowing it doesn't sort in-place, I some how switched to think it was in place..! - just kept `backingData` as unsorted as it doesn't really need to be, but sorted the actual display data